File: java/com/goodcrypto/crypto/CryptoException.java
package com.goodcrypto.crypto;

/**
 * Crypto exception.  
 * <p>Copyright 2003-2004 GoodCrypto
 * <br>2004.10.02
 * 
 * @version 0.1
 * @author GoodCrypto
 */
public final class CryptoException extends Exception
{
    public CryptoException () {
        super ();
    }
    
    public CryptoException (String message) {
        super (message);
    }
    
    public CryptoException (String message, Throwable cause) {
        super (message, cause);
    }
    
    public CryptoException (Throwable cause) {
        super (cause);
    }
}

File: python/goodcrypto/oce/crypto_exception.py
'''
    Copyright 2014-2016 GoodCrypto
    Last modified: 2016-06-10

    This file is open source, licensed under GPLv3 <http://www.gnu.org/licenses/>.
'''

class CryptoException(Exception):
    ''' Crypto exception.  '''

    def __init__(self, value=None):
        '''
            Constructor for the CryptoException.

            >>> message = 'oops'
            >>> try:
            ...     raise CryptoException('oops')
            ...     fail()
            ... except CryptoException as message_exception:
            ...     str(message_exception) == message
            True

            >>> try:
            ...     raise CryptoException()
            ... except CryptoException as exception:
            ...     print(exception.value)
            None
        '''

        if value is None:
            super(CryptoException, self).__init__()
        else:
            super(CryptoException, self).__init__(value)

        self.value = value

    def __str__(self):
        '''
            Get the string representation of the exception.

            >>> crypto_exception = CryptoException()
            >>> str(crypto_exception)
            'None'

            >>> crypto_exception = CryptoException('error message')
            >>> isinstance(crypto_exception.__str__(), str)
            True
        '''

        return str(self.value)

